Source/Particles/Deposition/VarianceAccumulationBuffer.cpp

```diff
@@ -0,0 +1,210 @@
+/* Copyright 2025 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: S. Eric Clark (Helion Energy)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "Particles/Deposition/VarianceAccumulationBuffer.H"
+#include "Particles/Deposition/TemperatureDeposition.H"
+#include "Parallelization/WarpXSumGuardCells.H"
+#include "Fields.H"
+
+#include <ablastr/utils/Communication.H>
+
+#include <AMReX.H>
+#include <AMReX_REAL.H>
+
+using namespace amrex::literals;
+using namespace warpx::particles::deposition;
+
+VarianceAccumulationBuffer::VarianceAccumulationBuffer (ablastr::fields::MultiLevelVectorField const& T_vf, std::string const& species_name ) :
+    m_species_name(species_name)
+{
+    using ablastr::fields::Direction;
+    auto & warpx = WarpX::GetInstance();
+
+    const int ncomps = 1;
+
+    for (int lev = 0; lev <= warpx.finestLevel(); ++lev) {
+        for (int idir = 0; idir < 3; ++idir) {
+            amrex::BoxArray const& ba = T_vf[lev][Direction{idir}]->boxArray();
+            amrex::DistributionMapping const& dm = T_vf[lev][Direction{idir}]->DistributionMap();
+            amrex::IntVect const& ng = T_vf[lev][Direction{idir}]->nGrowVect();
+
```

Comment:
Maybe use names that are more specific, something like "variance_w".

Reply:
This has been updated to prepend "variance_buffer_" to the field name for the register.